client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Route, RouteProps, useHistory } from 'react-router-dom';
+import { useAuth } from '../../context/useAuthContext';
+import { CircularProgress } from '@mui/material';
+
+const ProtectedRoute: React.FC<RouteProps> = ({ ...rest }) => {
```

Comment:
```suggestion
const ProtectedRoute = ({ component, path, ...rest }: RouteProps) : JSX.Element  => {
```
Let's also destructure some of the props :)

Short list of React.FC cons:
- Provides an implicit definition of children, even if your component doesn't need to have children. That might cause an error.
- Doesn't support generics.
- Doesn't work correctly with defaultProps.

Additional Reference: https://github.com/facebook/create-react-app/pull/8177